File: lab06/all_whales.pl
#!/usr/bin/perl

@array=();
%h=();
%hh=();
while ($a=<STDIN>){
$a=~ s/([A-Z])/\L$1\E/g;
$a=~ s/[\n\r]//g;
$a=~ s/ +$//;
if($a=~ /.*s$/){
chop($a);
}
$a=~ s/( )+/ /g;

$b=$a;
$b=~ s/^\d+ //g;
$name=$b;
if ($name~~@array){
$h{$name}++;
$a=~ /^(\d+) /;
$num=$1;
$hh{$name}=$hh{$name}+$num;

}
else{
push(@array, $name);
$h{$name}=1;
$a=~ /^(\d+) /;
$num=$1;
$hh{$name}=$num;

}
}
@w= sort @array;
foreach $whale(@w){
print "$whale observations: $h{$whale} pods, $hh{$whale} individuals\n";

}

File: lab06/all_whales.py
#!/usr/bin/python

import sys
import re


names=[]
h={}
hh={}

for line in sys.stdin:
    line=line.lower()
    lines=' '.join(line.split())
    if lines[-1]=='s':
        lines=lines[0:-1]
    name=re.findall(r"^\d+ (.*)",lines)
    if name[0] not in names:
        names.append(name[0])
        h[name[0]]=1
        shu=re.findall(r"^(\d+) .*",lines)
        hh[name[0]]=int(shu[0])
    else:
        h[name[0]]+=1
        shu=re.findall(r"^(\d+) .*",lines)
        hh[name[0]]+=int(shu[0])
names.sort()
for i in names:
    print("%s observations: %d pods, %d individuals" %(i, h[i], hh[i]))
    

        
    
